web-ng/src/app/services/auth/auth.guard.ts

```diff
@@ -14,36 +14,55 @@
  * limitations under the License.
  */
 import { Injectable } from '@angular/core';
-import { Router } from '@angular/router';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  Router,
+  RouterStateSnapshot,
+} from '@angular/router';
 import { AuthService } from './auth.service';
 import { Observable, of } from 'rxjs';
 import { catchError, map } from 'rxjs/operators';
 import { environment } from '../../../environments/environment';
+import { User } from '../../shared/models/user.model';
 
 @Injectable({
   providedIn: 'root',
 })
-export class AuthGuard {
-  private static readonly SIGN_IN_URL = '/signin';
-
+export class AuthGuard implements CanActivate {
   constructor(private router: Router, private authService: AuthService) {}
 
-  canActivate(): Observable<boolean> {
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ): Observable<boolean> {
     return this.authService.getUser$().pipe(
-      map(response => {
-        if (environment.useEmulators) {
-          return true;
+      map(user => {
+        if (state.url === '/signin') {
+          return this.handleSignin(user);
```

Comment:
Good effort on refactoring this logic into multiple functions; in this case, the added indirection makes the logic harder to follow because a) the method names `handleSigin` and `isAuthenticated` don't accurately reflect what the methods do and b) the logic being implemented is simpler than the refactor would lead on.